src/darker/__main__.py

```diff
@@ -572,9 +573,14 @@ def main(  # pylint: disable=too-many-locals,too-many-branches,too-many-statemen
     else:
         # In other modes, only reformat files which have been modified.
         if git_is_repository(root):
-            changed_files_to_reformat = git_get_modified_python_files(
-                files_to_process, revrange, root
-            )
+            # Get the modified files only.
+            repo_root = darker.black_compat.find_project_root([str(root)])
+            modified_files = git_get_modified_python_files(paths, revrange, repo_root)
+            changed_files = {
+                (repo_root / file).relative_to(root) for file in modified_files
+            }
```

Comment:
I admit the original code is confusing since it uses the words "modified" and "changed" interchangeably.

But I think this now adds further confusion by storing in
- `modified_files` the relative paths of modified files starting from the common root of given paths
- `changed_files` the relative paths of modified files starting from the Git repository root

Maybe this could be simplified to:
```suggestion
            changed_files = {
                (repo_root / file).relative_to(root)
                for file in
                git_get_modified_python_files(paths, revrange, repo_root)
            }
```